### PR TITLE
Introduce NoOp strategy for Application in case of pending migrations

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[db migrations](https://tools.hmcts.net/jira/browse/RPE-266)

### Change description ###

As part of CNP pipeline, applications should not run any migrations. There is a separate gradle task introduced in #47.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/config/FlywayConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.sendletter.database.migration.FlywayNoOpStrategy;
+
+import javax.sql.DataSource;
+
+@Configuration
+@ConditionalOnClass(Flyway.class)
+@ConditionalOnBean(DataSource.class)
+@ConditionalOnProperty(prefix = "flyway", name = "enabled", matchIfMissing = true)
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/config/FlywayConfiguration.java]: so with `flyway.enabled: true` Spring will run real migrations?

File: src/test/java/uk/gov/hmcts/reform/sendletter/SampleData.java [hunk 2]
```diff
@@ -32,6 +37,56 @@ public static String uuid() {
...
+            public Integer getExecutionTime() {
+                return null;
+            }
+
+            @Override
+            public int compareTo(MigrationInfo o) {
+                return 0;
+            }
+        };
+    }
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/sendletter/SampleData.java]: can we get away with just using a mock in tests?

**Comment by rkondratowicz** [comment]
just a random thought. this could go to https://github.com/hmcts/reform-api-standards (later)

File: src/main/java/uk/gov/hmcts/reform/sendletter/config/FlywayConfiguration.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/config/FlywayConfiguration.java]: spring will not configure flyway at all. this is a take out from spring's `FlywayAutoConfiguration`

File: src/test/java/uk/gov/hmcts/reform/sendletter/SampleData.java (hunk 2 above)
Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/sendletter/SampleData.java]: was thinking of using mocks but got all lazy implementing 3 or 4 mocks returning migration info

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/sendletter/SampleData.java]: not sure why it looked more hassle

**Pull request merged**